### PR TITLE
Adds a very basic weeaboo mini-antag

### PR DESCRIPTION
**Pull request by SonixApache**
revives #19689 except as weeaboo instead of ninja
now with dynamic rule entries

🆑 
 - rscadd: Nanotrasen has angered the local anime club by buying out their club room to build more space stations. As such, please beware of angry weeaboos.

### CODE REVIEW

File: code/game/objects/items/weapons/storage/backpack.dm [hunk 1]
```diff
@@ -198,3 +198,8 @@
 	name = "security messenger bag"
 	desc = "A tactical backpack worn over one shoulder. This one is in Security colors."
 	icon_state = "courierbagsec"
+
+/obj/item/weapon/storage/backpack/messenger/black
+	name = "black messenger bag"
+	desc = "for the freerunning pizza delivery hacker on the go."
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/storage/backpack.dm]: Capital F in "for."

**Review by Shadowmech88 — CHANGES_REQUESTED** [review]
<s>Being that this is a proof-of-concept/demonstration antag, some more comments explaining things like the difference between latejoin and midround rulesets and why they exist, the meanings of the `weight`, `cost`, and `requirements` vars, and other things that may not be obvious to those who have no experience with antag datums, would be appreciated.</s>
The documentation included with the definitions of the vars and procs seems sufficient. It's just the typo, in that case.

**Comment by SonixApache** [comment]
@Shadowmech88 
There's a boatload of documentation in these two files, which helped me set this up.
https://github.com/vgstation-coders/vgstation13/blob/2b7409e03056c1fc41fa2e844136eca426e44ecb/code/datums/gamemode/dynamic/dynamic_rulesets.dm#L3-L17
https://github.com/vgstation-coders/vgstation13/blob/2b7409e03056c1fc41fa2e844136eca426e44ecb/code/datums/gamemode/role/role.dm#L1-L3
Do you think more is necessary?

**Review by Shadowmech88 — APPROVED** [review]

File: code/modules/mob/inventory.dm [hunk 2]
```diff
@@ -279,20 +279,25 @@
 	return 0
 
 
-/mob/proc/drop_from_inventory(var/obj/item/W) //I'm fairly sure the entirety of this proc is redundant and can be replaced by just u_equip(W,1)
+/mob/proc/drop_from_inventory(var/obj/item/W, var/delete = FALSE) //I'm fairly sure the entirety of this proc is redundant and can be replaced by just u_equip(W,1)
```
Comment (DamianX) [inline code/modules/mob/inventory.dm]: why?

Comment (Exxion) [reply code/modules/mob/inventory.dm]: In addition to being stupid, this change is, as far as I can tell, unused

Comment (SonixApache) [reply code/modules/mob/inventory.dm]: misc_gamemode_procs.dm#L276 calls `H.drop_all(TRUE)`
`drop_all()` calls `drop_from_inventory` for every item in the inventory
I guess @MadmanMartian forgot a line. I'll fix it.

Comment (MadmanMartian) [reply code/modules/mob/inventory.dm]: equip_weeaboo proc uses it, through drop_all

Comment (SonixApache) [reply code/modules/mob/inventory.dm]: Did it pass the var before I added that line?

Comment (DamianX) [reply code/modules/mob/inventory.dm]: I'd just make it a separate proc that just `qdel`s everything equipped, `item/Destroy()` already calls `drop_from_inventory` if necessary

**Comment by MadmanMartian** [comment]
Just had the best idea:

> 50% chance the weeaboo has cat ears equipped

File: code/modules/mob/inventory.dm (hunk 2 above)
Comment (SonixApache) [reply code/modules/mob/inventory.dm]: Alright, I'll make a seperate proc.

**Comment by tralezab** [comment]
this pr is awesome, here's a child of the katana that lets weeaboos teleport behind someone they attack every one minute if you want to use it

https://pastebin.com/rtW4JwkV

**Comment by SonixApache** [comment]
@tralezab looks cute, I'll think about it. Your alt-click just flips the teleport var, though, might want to fix that. Could you just make a PR on the `crazed_weeaboo` branch on my fork?

**Comment by tralezab** [comment]
sure thing, and i'll fix the alt-click stuff. i really did write that in pastebin so I kinda expected issues 😅 

and yeah this time i'll compile it to make sure the code works and all that

**Comment by SonixApache** [comment]
Going to fix the macros / logos. ~~Do I still need to make that proc, or is how it's currently set up fine?~~

**Comment by SonixApache** [comment]
Should be good for review round 2.

File: vgstation13.dme [hunk 3]
```diff
@@ -2430,7 +2430,7 @@
 #include "maprendering\maprendering.dm"
 #include "maps\_map.dm"
 #include "maps\_map_override.dm"
-#include "maps\tgstation.dm"
+#include "maps\test_tiny.dm"
```
Comment (MadmanMartian) [inline vgstation13.dme]: Oof

Comment (SonixApache) [reply vgstation13.dme]: goddamnit

**Comment by SonixApache** [comment]
conflicts removed and such

**Comment by ShiftyRail** [comment]
You can actually make a pref for it rather easily, just add it to the prefs selection screen and make a few defines for it.
Adding it to the ban database is a bit trickier, but you could just use the `role_category_override` to imply that it shares a banlist with traitors.

**Comment by Intigracy** [comment]
confrictu

**Review by Intigracy — DISMISSED** [review]
If you're going to have it be based on the regular traitor pref, that's fine.

Just fix the conflicts and I'll merge it.

**Comment by SonixApache** [comment]
it is done

File: code/datums/gamemode/objectives/target/syndicate/skulls.dm [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/datum/objective/target/skulls
+	name = "\[Syndicate\] steal skulls"
+	var/amount
+
+/datum/objective/target/skulls/format_explanation()
+	return "Capture [amount] trophy skulls. They must be from NT employees."
```
Comment (MadmanMartian) [inline code/datums/gamemode/objectives/target/syndicate/skulls.dm]: Might be best to change it to "trophies (decapitated heads)" rather than skulls

Comment (SonixApache) [reply code/datums/gamemode/objectives/target/syndicate/skulls.dm]: i was looking at that line myself a while ago, you're probably right

**Comment by SonixApache** [comment]
confricto resovored againu

**Comment by SonixApache** [comment]
going to add a pref now that i've seen how easy it is

File: code/datums/gamemode/role/weeaboo.dm [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/datum/role/weeaboo
+	name = WEEABOO 
+	id = WEEABOO
```
Comment (SonixApache) [inline code/datums/gamemode/role/weeaboo.dm:3]: This line needs to be set to `ROLE_WEEABOO` for the polling and rule executing to actually work. However, this breaks the macro, and isn't how it's supposed to work if you look at all the other roles.

I having a hard time figuring out how to resolve this.

**Comment by SonixApache** [comment]
tested, it all works now
not pretty but hey whatever

**Pull request merged**